test/e2e/end_user_test.go

```diff
@@ -163,4 +135,99 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 		fmt.Println(result.Error().Error())
 		Expect(result.Error().Error()).To(ContainSubstring("pods \"sync-master-000000\" is forbidden: User \"enduser\" cannot get pods/log in the namespace \"kube-system\""))
 	})
+
+	It("should deploy a template with persistent storage and test failure modes", func() {
+		tpl := "django-psql-persistent"
+		By(fmt.Sprintf("instantiating the template and getting the route (%v)", time.Now()))
+		// instantiate the template
+		err := c.instantiateTemplate(tpl)
+		Expect(err).NotTo(HaveOccurred())
+		// Pull the route ingress from the namespace
+		route, err := c.rc.Routes(c.namespace).Get(tpl, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		// make sure only 1 ingress point is returned
+		Expect(len(route.Status.Ingress)).To(Equal(1))
+		host := route.Status.Ingress[0].Host
+		url := fmt.Sprintf("http://%s", host)
+
+		// Curl the endpoint and search for a string
+		httpc := &http.Client{
+			Timeout: 10 * time.Second,
+		}
+		var regex = regexp.MustCompile(`Page views:\s*(\d+)`)
```

Comment:
nit: I'd prefer it stylistically if you did `foo :=` consistently instead of `var foo =`